transports/tcp/src/lib.rs

```diff
@@ -734,116 +721,88 @@ where
     T: Provider,
     T::Listener: Unpin,
     T::Stream: Unpin,
-    T::IfWatcher: Unpin,
 {
     type Item = Result<TcpListenerEvent<T::Stream>, io::Error>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
         let me = Pin::into_inner(self);
 
-        loop {
-            match &mut me.in_addr {
-                InAddr::Any { if_watch, addrs } => match if_watch {
-                    // If we listen on all interfaces, wait for `if-watch` to be ready.
-                    IfWatch::Pending(f) => match ready!(Pin::new(f).poll(cx)) {
-                        Ok(w) => {
-                            *if_watch = IfWatch::Ready(w);
-                            continue;
+        match &mut me.in_addr {
+            InAddr::Any(if_watcher) => {
+                while let Poll::Ready(ev) = if_watcher.as_mut().poll_if_event(cx) {
```

Comment:
I don't think we need `as_mut` here.

```suggestion
                while let Poll::Ready(ev) = if_watcher.poll_if_event(cx) {
```

Reply:
If we would use a `loop` here, there would be only one `match` that includes the `Poll` which again, would be slightly more consistent with the remaining blocks in this function.

Reply:
Done with 8d16ad7936ef3fb74bd58161761a4139f9550fc5.